docs/source/Mapping.rst

```diff
@@ -0,0 +1,152 @@
+
+Consider the following circuit.
+
+    .. code-block:: python3
+
+        from mqt import qmap
+        from qiskit import QuantumCircuit
+
+        qc = QuantumCircuit(4)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.cx(0, 2)
+        qc.cx(0, 3)
+
+        qc.barrier()
+
+        qc.t(0)
+        qc.t(1)
+        qc.t(2)
+        qc.t(3)
+
+        qc.barrier()
+
+        qc.cx(0, 3)
+        qc.cx(0, 2)
+        qc.cx(0, 1)
+
+        qc.measure_all()
+
+        print(qc.draw(fold=-1))
+
+    .. code-block:: console
+
+              ┌───┐                ░ ┌───┐ ░                 ░ ┌─┐
+         q_0: ┤ H ├──■────■────■───░─┤ T ├─░───■────■────■───░─┤M├─────────
+              └───┘┌─┴─┐  │    │   ░ ├───┤ ░   │    │  ┌─┴─┐ ░ └╥┘┌─┐
+         q_1: ─────┤ X ├──┼────┼───░─┤ T ├─░───┼────┼──┤ X ├─░──╫─┤M├──────
+                   └───┘┌─┴─┐  │   ░ ├───┤ ░   │  ┌─┴─┐└───┘ ░  ║ └╥┘┌─┐
+         q_2: ──────────┤ X ├──┼───░─┤ T ├─░───┼──┤ X ├──────░──╫──╫─┤M├───
+                        └───┘┌─┴─┐ ░ ├───┤ ░ ┌─┴─┐└───┘      ░  ║  ║ └╥┘┌─┐
+         q_3: ───────────────┤ X ├─░─┤ T ├─░─┤ X ├───────────░──╫──╫──╫─┤M├
+                             └───┘ ░ └───┘ ░ └───┘           ░  ║  ║  ║ └╥┘
+      meas: 4/══════════════════════════════════════════════════╩══╩══╩══╩═
+                                                          0  1  2  3
+
+
+Now assume this circuit shall be mapped to a 4-qubit architecture with the following coupling map:
+
+.. image:: /images/linear_arch_4-1.png
+   :width: 80%
+   :alt: 4-qubit Architecture
+   :align: center
+
+
+In *QMAP* this architecture can be manually defined.
+
+    .. code-block:: python3
+
+        arch = qmap.Architecture(4, set([(0, 1), (1, 2), (2, 3), (3, 2), (2, 1), (1, 0)]))
+
+The quantum circuit :code:`qc` can not be run directly on this architecture. Naive mapping yields the following possible compiled circuit.
+
+    .. code-block:: console
+
+                      ┌───┐                      ░ ┌───┐ ░                       ░ ┌─┐
+	  q_0 -> q_0: ┤ H ├──■───X───────────────░─┤ T ├─░───────────────X───■───░─┤M├─────────
+	              └───┘┌─┴─┐ │               ░ ├───┤ ░               │ ┌─┴─┐ ░ └╥┘┌─┐
+	  q_1 -> q_1: ─────┤ X ├─X───■───X───────░─┤ T ├─░───────X───■───X─┤ X ├─░──╫─┤M├──────
+	                   └───┘   ┌─┴─┐ │       ░ ├───┤ ░       │ ┌─┴─┐   └───┘ ░  ║ └╥┘┌─┐
+	  q_2 -> q_2: ─────────────┤ X ├─X───■───░─┤ T ├─░───■───X─┤ X ├─────────░──╫──╫─┤M├───
+	                           └───┘   ┌─┴─┐ ░ ├───┤ ░ ┌─┴─┐   └───┘         ░  ║  ║ └╥┘┌─┐
+	  q_3 -> q_3: ─────────────────────┤ X ├─░─┤ T ├─░─┤ X ├─────────────────░──╫──╫──╫─┤M├
+	                                   └───┘ ░ └───┘ ░ └───┘                 ░  ║  ║  ║ └╥┘
+	      meas: 4/══════════════════════════════════════════════════════════════╩══╩══╩══╩═
+                                                                                    0  1  2  3
```

Comment:
I am not quite sure if the indentation here is right. at least GitHub displays it in a weird fashion.

Reply:
Typical case where Tabs will ruin your life.